executor/src/witgen/jit/compiler.rs

```diff
@@ -412,21 +405,17 @@ fn format_effect<T: FieldElement>(effect: &Effect<T, Variable>, is_top_level: bo
                 "".to_string()
             };
 
-            if matches!(second[..], [Effect::Branch(..)]) {
-                format!(
-                    "{var_decls}if {} {{\n{}\n}} else if {}",
```

Comment:
The actual fix is to change `else if` to just `else` here (the inner branch will create another `if`). The rest is just some code rearranging - sorry, couldn't help it.